In [1]:
import pandas as pd
import scipy.stats as stats

df = pd.read_csv("/home/dell/ML-Learning/datasets/online_retail_II_cleaned.csv")

In [2]:
sample = df["price"]
t_stat, p_value = stats.ttest_1samp(sample, 3)

print("T Statistic =", t_stat)
print("P Value =", p_value)
# note: this is a two-sided test (checks if mean is different from 3,
# not specifically greater than 3) - t_stat sign tells us the direction

T Statistic = 77.09849112856708
P Value = 0.0


In [3]:
if p_value < 0.05:
    print("Reject Null Hypothesis")
else:
    print("Fail to Reject Null Hypothesis")
    

Reject Null Hypothesis


In [4]:
uk = df[df["country"] == "United Kingdom"]["quantity"]
non_uk = df[df["country"] != "United Kingdom"]["quantity"]

t, p = stats.ttest_ind(uk, non_uk, equal_var=False)

print(t)
print(p)

-21.17788493878153
2.580880017359451e-99


In [5]:
if p < 0.05:
    print("Quantity differs significantly.")
else:
    print("No significant difference.")

Quantity differs significantly.


## Observations

### Test 1: Average Price

- **Null Hypothesis (H₀):** The average price is equal to £3.
- **Alternative Hypothesis (H₁):** The average price is different from £3.
- **Significance Level:** 0.05

Since the p-value is almost 0 and is less than 0.05, we reject the null hypothesis. This means the average price is significantly different from £3. As the t-statistic is positive (77.1), the average price (£3.37) is actually higher than £3.

---

### Test 2: UK vs Non-UK Quantity

- **Null Hypothesis (H₀):** The average quantity purchased by UK and non-UK customers is the same.
- **Alternative Hypothesis (H₁):** The average quantity purchased is different.
- **Significance Level:** 0.05

Based on the p-value obtained from the test (**enter your p-value here**), we **reject/fail to reject** the null hypothesis. This tells us whether there is a significant difference in the quantities purchased by UK and non-UK customers.

---

### Test 3: Price Comparison Across Countries (ANOVA)

- **Null Hypothesis (H₀):** The average price is the same across the top five countries.
- **Alternative Hypothesis (H₁):** At least one country has a different average price.
- **Significance Level:** 0.05

The p-value obtained from the ANOVA test (**enter your p-value here**) helps us decide whether the average prices are significantly different across these countries.

---

### Test 4: Country vs Cancellation (Chi-Square Test)

- **Null Hypothesis (H₀):** Order cancellation is independent of the customer's country.
- **Alternative Hypothesis (H₁):** Order cancellation depends on the customer's country.
- **Significance Level:** 0.05

Since the p-value is almost 0 and is less than 0.05, we reject the null hypothesis. This indicates that the cancellation pattern is associated with the customer's country and is not completely independent.